### PR TITLE
Make Ember.compare public

### PR DESCRIPTION
**Pull request by alexspeller**
more method visibility fixes in the `Ember` namespace

### CODE REVIEW

File: packages/ember-metal/lib/keys.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
   @for Ember
   @param {Object} obj
   @return {Array} Array containing keys of obj
-  @private
+  @public
```
Comment (rwjblue) [inline packages/ember-metal/lib/keys.js]: This was deprecated in 1.13 in favor of using Object.keys. This whole file should be removed (TBH, I thought it was already).

Comment (alexspeller) [inline packages/ember-metal/lib/keys.js]: @rwjblue Should I change the PR to remove the file, or should I change it to just deal with `Ember.compare`?

Comment (rwjblue) [inline packages/ember-metal/lib/keys.js]: Lets make it `[DOC release]` and just deal with `Ember.compare`.  We can remove the file in a separate `[CLEANUP beta]` PR if you don't mind...

**Comment by rwjblue** [comment]
Restarted sauce labs job

**Pull request merged**